attachement2Component.js

```diff
@@ -0,0 +1,55 @@
+import { LightningElement, api } from "lwc";
+import uploadFile from "@salesforce/apex/FileUpload.uploadFile";
+import generatePublicLink from "@salesforce/apex/GeneratePublicUrl.generatePublicLink";
+
+export default class Attachement2Component extends LightningElement {
+  @api recordId;
+  DocumentId;
+  fileData;
+  publicUrl;
+  error;
+
+  openfileUpload(event) {
+    const file = event.target.files[0];
+    let reader = new FileReader();
+    reader.onload = () => {
+      let base64 = reader.result.split(",")[1];
+      this.fileData = {
+        filename: file.name,
+        base64: base64,
+        recordId: this.recordId,
+      };
+    };
+    reader.readAsDataURL(file);
+  }
```

Comment:
you should make your function process on array of files, not one file at a time.